src/content/containers/ProfileThreadHeaderContextMenu.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { ContextMenu, MenuItem } from 'react-contextmenu';
+import actions from '../actions';
+import { connect } from 'react-redux';
+import { getThreads, getThreadOrder } from '../reducers/profile-view';
+import { getFriendlyThreadName } from '../profile-data';
+import classNames from 'classnames';
+
+import type { Thread, ThreadIndex } from '../../common/types/profile';
+import type { Action } from '../actions/types';
+
+type Props = {
+  contextMenuId: string,
+  threads: Thread,
+  threadOrder: ThreadIndex[],
+  hideThread: ThreadIndex => Action,
+  showThread: ThreadIndex => Action,
+}
+
+class ProfileThreadHeaderContextMenu extends PureComponent {
+
+  constructor(props: Props) {
+    super(props);
+    (this: any).handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(
+    event: SyntheticEvent,
```

Comment:
Looking into it I guess technically it's a `SyntheticMouseEvent | SyntheticTouchEvent`, lol.